types/mithril/index.d.ts

```diff
@@ -1,863 +1,268 @@
-		* @see m.component
-		*/
-		controller: ControllerFunction<T> | ControllerConstructor<T>;
-
-		/**
-		* Creates a view out of virtual elements.
-		*
-		* @see m.component
-		*/
-		view(ctrl?: T, ...args: any[]): VirtualElement;
+	export interface JsonpOptions {
+		/** The data to be interpolated into the URL and serialized into the querystring. */
+		data?: any;
+		/** A constructor to be applied to each object in the response. */
+		type?: new (o: any) => any;
+		/** The name of the function that will be called as the callback. */
+		callbackName?: string;
+		/** The name of the querystring parameter name that specifies the callback name. */
+		callbackKey?: string;
+		/** If false, redraws mounted components upon completion of the request. If true, it does not. */
+		background?: boolean;
 	}
 
-	/**
-	* This is the base interface for property getter-setters
-	*
-	* @see m.prop
-	*/
-	interface Property<T> {
-		/**
-		* Gets the contained value.
-		*
-		* @return The contained value.
-		*/
-		(): T;
-
-		/**
-		* Sets the contained value.
-		*
-		* @param value The new value to set.
-		* @return The newly set value.
-		*/
-		(value: T): T;
+	export interface JsonpOptionsAll extends JsonpOptions {
+		/** The URL to send the request to. */
+		url: string;
 	}
 
-	/**
-	* This represents a non-promise getter-setter functions.
-	*
-	* @see m.prop which returns objects that implement this interface.
-	*/
-	interface BasicProperty<T> extends Property<T> {
-		/**
-		* Makes this serializable to JSON.
-		*/
-		toJSON(): T;
+	export interface Jsonp {
+		/** Makes a JSON-P request and returns a promise. */
+		<T>(options: JsonpOptionsAll): Promise<T>;
+		/** Makes a JSON-P request and returns a promise. */
+		<T>(url: string, options?: JsonpOptions): Promise<T>;
 	}
 
-	/**
-	* This represents a key-value mapping linking routes to components.
-	*/
-	interface Routes {
-		/**
-		* The key represents the route. The value represents the corresponding
-		* component.
-		*/
-		[key: string]: Component<Controller>;
+	export interface RequestService {
+		request: Request;
+		jsonp: Jsonp;
 	}
 
-	/**
-	* This represents a Mithril deferred object.
-	*/
-	interface Deferred<T> {
-		/**
-		* Resolve this deferred's promise with a value.
-		*
-		* @param value The value to resolve the promise with.
-		*/
-		resolve(value?: T): void;
-
-		/**
-		* Reject this deferred with an error.
-		*
-		* @param value The reason for rejecting the promise.
-		*/
-		reject(reason?: any): void;
-
-		/**
-		* The backing promise.
-		*
-		* @see Promise
-		*/
-		promise: Promise<T>;
+	/** Renders a vnode structure into a DOM element. */
+	export type Render = (el: Element, vnodes: Children) => void;
+
+	export interface RenderService {
+		render: Render
 	}
 
-	/**
-	* This represents a thennable success callback.
-	*/
-	interface SuccessCallback<T, U> {
-		(value: T): U | Thennable<U>;
+	/** Manually triggers a redraw of mounted components. */
+	export type Redraw = () => void;
```

Comment:
Move this to `mithril/redraw.d.ts` as `export function redraw(): void`, and ensure you have the following other export:

```ts
export {render} from './render';
```